### PR TITLE
Rightsize resources for production

### LINKED ISSUE #63: Rightsize Azure Resources for Production
Many of the Azure resources have currently been sized on small / basic SKUs. This task is to track the need to revisit them for a productionised deployment. 

- [ ] SQL Database
- [ ] Databricks cluster settings
- [ ] ...

### PR DESCRIPTION
**Pull request by t-young31**
# Resolves #63 

### Resources considered

- ACR: Basic sku, 10 GB included storage, pay per GB
- ASP: Standard sku, has auto scaling
- CosmosDB: Serverless thus scalable. Recommended for low traffic scenarios, which FlowEHR apps will be 
- Databricks: see below
- SQL server: Basic, upped to standard in production environments where the 250 GB upper storage limit should be suitable
- Storage accounts: Standard + GRS replication, seems reasonable(?)

### Has the CIS document been consulted / updated?

> - Are you adding new resources, or modifying existing ones? 
> - Have settings been set in line with the CIS recommendations?

Modifying. Yes

***

### Questions

- Is it worth running app dev pipelines against a more realistic size of data so we know if we need to scale out the databricks cluster?
- Do we keep the Basic SQL server for dev environments? (thinking #224 ). I'd vote maybe because of the price difference – perhaps something worth discussing?

### CODE REVIEW

**Comment by jjgriff93** [comment]
1. Yes think that's a sensible idea - do we have / can we get a large dataset relatively easily?
2. I'd vote we have the same SQL instance for both if we have pipelines that will be churning through representative amounts of synthetic data for testing - or we at least make it configurable from the config.yaml so that it can be set to basic for local scenarios, but for app dev for example it can be rightsized

**Comment by tanya-borisova** [comment]
1. Yes, that sounds good, would it just be a matter of upscaling the synthetic data generator?

**Comment by damoodamoo** [comment]
1. Running app-dev against better data size - yes would be nice, but as @tanya-borisova said is not always going to give us something we could go on (especially if a lot of data is selected straight out as it's synthetic). Best bet is to run on ML-Dev and see how we go, even manually tuning cluster size and re-running there over and over.
2. We need the limit to be more than 2GB even in dev, so whatever the next cheapest thing that has a bigger limit is good. On this, for prod - 250GB doesn't sound huge - can we confirm that if we get to that limit that we can just raise the SKU and carry on growing?

File: infrastructure/transform/feature-data-store.tf [hunk 1]
```diff
@@ -101,9 +101,9 @@ resource "azurerm_mssql_database" "feature_database" {
   server_id            = azurerm_mssql_server.sql_server_features.id
   collation            = "SQL_Latin1_General_CP1_CI_AS"
   license_type         = "LicenseIncluded"
-  max_size_gb          = 2
-  sku_name             = "Basic"
-  storage_account_type = "Local"
+  max_size_gb          = var.accesses_real_data ? 250 : 2
```
Comment (damoodamoo) [inline infrastructure/transform/feature-data-store.tf:104]: may as well make it 250GB / S0 for all.

**Review by damoodamoo — APPROVED** [review]

**Review by jjgriff93 — APPROVED** [review]

**Comment by jjgriff93** [comment]
:robot: pr-bot :robot:

:runner: Running tests: https://github.com/UCLH-Foundry/FlowEHR/actions/runs/4698343870 (with refid `0acd30e5`)

(in response to [this comment](https://github.com/UCLH-Foundry/FlowEHR/pull/239#issuecomment-1508213380) from @t-young31)

**Pull request merged**